rpcapd/daemon.c

```diff
@@ -1234,6 +1234,11 @@ daemon_msg_auth_req(struct daemon_slpars *pars, uint32 plen)
 			uint32 usernamelen, passwdlen;
 
 			usernamelen = ntohs(auth.slen1);
+			if (usernamelen >= 0xFFFFFFFF) {
```

Comment:
`usernamelen` is set by a call to `ntohs()`, which puts a 16-bit quantity into host byte order from network byte order; as such, it should never produce a result > 65535.  How would it be possible for `usernamelen` to be > 65535?